interview-questions.md

```diff
@@ -8,50 +8,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is variable that's used in a function. An argument is the same thing, except its used when you are calling on the parameter. 
 
-Researched answer:
+Researched answer: Parameters are like placeholders in a function. The arguments are the actual data that replaces the parameters in a function when it is being called on.
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: .map requires index. The order is always index, value, array and must be in order, even when the prior ones aren't being used. value and array are optional.
 
-Researched answer:
+Researched answer: .map can also be used to reformat objects in an array.
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: .map will output the exact same number of objects in the array, while .filter will only output a subset of the array based on what parameters are passed.
 
-Researched answer:
+Researched answer: .filter will filter through each element in the array and return a filtered array. Map will output a new array, with the same number of elements as the original array. There is also forEach and reduce. forEach does the same thing as filter, except forEach will execute the function on each element, while filter will remove the element if the return value is false. Reduce will reduce the entire array to a single value.
 
 1. What is iteration?
 
-Your answer:
+Your answer: Iteration is the process of calling on each element of an array. Using a for loop, we can tell the for loop what index value to start at, where to end at, and the rate of the value it increases each time the for loop is executed.
 
-Researched answer:
+example: for (let i = 0; array.length; i++)
+
+The first part is index where the for loop will first execute. The second part is telling it where to stop. In this case, it will execute until it reaches the end of the array. The last part is shorthand for telling it to increase the value by 1 each time it executes, essentially going through each element, one by one.
+
+Researched answer: There are a few other iteration methods, like do.. while. do and while used together will execute a function on each element until it returns false.
```

Comment:
I think you can benefit from some review of iteration! Iteration is the process of executing code over and over until a desired outcome is met. For loops and while loops are indeed ways to initiate iteration, but iteration also occurs in a lot of different methods and higher order functions. `.map`, `.filter`, `.forEach`, all iterate through what they are attached to. Other methods such as `.reduce`, `.search`... countless others all use iteration.